tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2639,8 +2637,99 @@ def visit_functiondef(self, node):
             pass
 
 
-class DoNotImportLegacySix(BaseChecker):
+class ImportTypeChecker(BaseChecker):
+    """Checker to ensure no type is imported from more than one module in the same package,
+    except for clients in sync/async namespaces."""
+
+    name = "import-type-checker"
+    priority = -1
+    msgs = {
+        "C4764": (
+            "Type %s is imported multiple times from different modules: %s and %s",
+            "duplicate-import-type",
+            "Used when a type is imported multiple times, even from different modules, within the same package.",
+        ),
+    }
+
+    def visit_module(self, node):
+        """Initialize the dictionary for tracking imports at the start of processing each file."""
+        self.imported_entities = {}
+        self.dunder_all_exports = set()
+
+    def visit_importfrom(self, node):
+        """Check for duplicate imports from different modules within the same package."""
+        if self._is_inside_exception_or_conditional(node):
+            # Ignore imports inside conditional or try/except blocks
+            return
+
+        module_name = node.modname
+        for name, alias in node.names:
+            entity_name = alias or name
+
+            # Check if the entity has already been imported
+            if entity_name in self.imported_entities:
+                # Special case for clients with sync/async namespaces
```

Comment:
Can you add a test for this scenario, I'm not sure I understand what the special case is?

Reply:
the scenario was for special cases for clients, who should have the same name between sync and async namespaces. If its not need i can remove it.

Reply:
I see, thanks! Both clients should have the [same name](https://azure.github.io/azure-sdk/python_design.html#python-client-same-name-sync-async) - I would recommend adding a test for it using a full module path i.e `from azure.eventgrid.aio ...... vs from azure.eventgrid ......` :)

Reply:
Special case test has been added